src/Directory.Build.targets

```diff
@@ -2,12 +2,30 @@
 <!-- Copyright (c) .NET Foundation and contributors. All rights reserved. Licensed under the MIT license. See License.txt in the project root for full license information. -->
 <Project>
 
-  <ItemGroup Condition="'$(IsTestProject)' == 'true' AND '$(OutputType)' == 'Exe'">
-    <Compile Include="$(MSBuildThisFileDirectory)Tests\Common\Program.cs" />
-  </ItemGroup>
-
+  <PropertyGroup>
+    <MSBuildAllProjects>$(MSBuildAllProjects);$(MSBuildThisFileFullPath)</MSBuildAllProjects>
+  </PropertyGroup>
+  
+  <PropertyGroup Condition="'$(IsTestProject)' == 'true' AND '$(OutputType)' == 'Exe'">
+    <IsPackable>true</IsPackable>
+    <PackAsTool>true</PackAsTool>
+    <ToolCommandName>$(PackageId)</ToolCommandName>
+    
+    <!-- Put packages for tests in subfolder so we don't try to sign them -->
+    <PackageOutputPath>$(PackageOutputPath)tests\</PackageOutputPath>
```

Comment:
We will need to modify the build definition to also push the NuGet packages in this folder.  (If we switch to YAML build definitions changes like this will be a bit easier)